site/content/en/docs/Reference/API/Kustomization File/configMapGenerator.md

```diff
@@ -6,3 +6,81 @@ weight: 6
 description: >
     Generate ConfigMap resources.
 ---
+
+## configMapGenerator
+
+ConfigMap objects can be generated by adding a `configMapGenerator` entry to the `kustomization.yaml` file.
+
+See [Generating ConfigMaps Tasks] for examples of how to use `configMapGenerator`.
+
+---
+
+* **apiVersion**: kustomize.config.k8s.io/v1beta1
+* **kind**: Kustomization
+* **configMapGenerator** ([][ConfigMapArgs](#configmapargs))
+
+    List of ConfigMaps to generate.
+
+
+### ConfigMapArgs
```

Comment:
I don't think `ConfigMapArgs`  is exposed to the user. It is an implementation detail. We might not want to document it, unless it's re-used by `secretGenerator` (can't remember off the top of my head, I think it's re-used at least partially). We can potentially document the re-used fields on a separate page.

Reply:
The ConfigMap and Secret generators use [type.ConfigMapArgs](https://github.com/kubernetes-sigs/kustomize/blob/master/api/types/configmapargs.go) and [type.SecretArgs](https://github.com/kubernetes-sigs/kustomize/blob/master/api/types/secretargs.go). These types share [type.GeneratorArgs](https://github.com/kubernetes-sigs/kustomize/blob/master/api/types/generatorargs.go) though they are different because the secret specifies a type.

 Do you have a suggestion for another way to represent this? We could specify a `GeneratorArgs` in the common definitions?

Reply:
I added `GeneratorArgs` to common definitions. Let me know if you are okay with this approach.

Reply:
I see these files refer specifically to the kustomization file. `configMapArgs`, `secretArgs`, and `generatorArgs` are all fields that the end user is not expected to have in their `kustomization.yaml`. I feel there might be a confusion potential if these are described in this section, but it is useful to have them described in a separate API section for people onboarding to the project as contributors.

Reply:
I am definitely open to suggestions for how best to describe the generator arguments. I was attempting to document that Secret and ConfigMap generators share many of the same arguments while avoiding listing the same fields in multiple places.

Using [env](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#environment-variables) in the PodSpec as an example, users are not expected to specify an `EnvVar` field. It is understood that `EnvVar` is a type used to define the fields to include in the `env` list.

Reply:
If this is using Hugo, we can create a shortcode for including partial pages inside of other pages, similar to what [docs.k8s.io](https://docs.k8s.io) currently does:
* Sample shortcode: https://github.com/kubernetes/website/blob/022634424d27cb1b3aa5ca0b45999977f5be3c9b/layouts/shortcodes/include.html
* Sample usage: https://github.com/kubernetes/website/blob/bc7cbf6e530acd3e4332cc257041e39a9f5e102c/content/en/docs/tasks/tools/install-kubectl-linux.md

Reply:
Thank you for sharing this hugo feature. Though I am not sure this addresses how to document a list of generator arguments. For example, the `configMapGenerator` field is a list of the `ConfigMapArgs` type. I recognize that users don't specify `configMapArgs` anywhere in the Kustomization file, but I believe there should be a way to document what arguments are available for each ConfigMap object in the list.

Reply:
Please disregard my previous comment. I used your suggestion to create a shortcode to include other markdown and I think the result is improved for the reader. Thank you again for the tip.

Reply:
I just checked the rendered preview and it looks nice! 🎉